packages/realm-react/src/useObject.tsx

```diff
@@ -55,24 +55,94 @@ export function createUseObject(useRealm: () => Realm) {
     // the cachedObject can force the component using this hook to re-render when a change occurs.
     const [, forceRerender] = useReducer((x) => x + 1, 0);
 
+    // Get the original object from the realm, so we can check if it exists
+    const originalObject = realm.objectForPrimaryKey<unknown>(type, primaryKey);
+
+    // Store the primaryKey as a ref, since when it is an objectId or UUID, it will be a new instance on every render
+    const primaryKeyRef = useRef(primaryKey);
```

Comment:
So what will happen if the calling code starts calling this hook with a different id? Do we have a test for that - if not, I'd suggest adding one.